#167749954 Implement Password Reset Feature

daviddamilola opened the pull request:
#### What does this PR do?
 Add password reset feature
#### Description of Task to be completed?
* add test for password reset endpoint
* add password reset endpoint routes.
* add update Password service.
* add verify email and send email Helper.
* add password validation user schema.
* update documentation.
#### How should this be manually tested?
* `git clone` the branch
* set the environment variables
* run postman
* send a `POST` request to ` /api/v1/users/reset`
* send a `PATCH`  request containing the new password to the link in the email
#### Any background context you want to provide?
`nodemailer` was used to send the email.
#### What are the relevant pivotal tracker stories?
[#167749954](https://www.pivotaltracker.com/story/show/167749954)

hustlaviola commented on .travis.yml:
```diff
@@ -9,6 +9,11 @@ before_script:
 node_js:
 - 'stable'
 
+env:
```
You can remove the environment variables, it has been added on Travis

daviddamilola replied:
changes have been implemented.

hustlaviola commented on src/config/swagger.json:
```diff
@@ -80,10 +193,19 @@
                 "type": "integer",
                 "format": "int64"
               },
+              "firstname": {
```
```suggestion
              "firstName": {
```

daviddamilola replied:
changes have been implemented.

hustlaviola commented on src/config/swagger.json:
```diff
@@ -80,10 +193,19 @@
+              "firstname": {
+                "type": "string"
+              },
+              "lastname": {
```
```suggestion
              "lastName": {
```

daviddamilola replied:
changes have been implemented

hustlaviola commented on src/controllers/UserController.js:
```diff
@@ -39,4 +41,41 @@ export default class UserController {
+      Responses.setError(400, 'this address link has expired')
+      return Responses.send(res);
+    }
+    if (req.method == 'GET') {
```
```suggestion
    if (req.method === 'GET') {
```

daviddamilola replied:
changes have been implemented

hustlaviola commented on src/database/models/user.js:
```diff
@@ -19,4 +20,4 @@ export default (sequelize, DataTypes) => {
   }, {});
   return User;
-};
+};
```
You can always add newlines at the end of a file

Naimaoye reviewed (changes requested):
Good job David. I think you need to add new lines at the end of some of your files, please check through

hustlaviola reviewed (commented):
A lot of files does not have newlines at the end. Consider adding them. You can also rebase off `ft-JWT-auth-167749949` to implement some changes on your branch. This is good work, well done.

hustlaviola commented on src/routes/index.js:
```diff
@@ -0,0 +1,11 @@
+import express from 'express';
```
This file has been deleted. Consider rebasing

hustlaviola commented on src/tests/index.test.js:
```diff
@@ -18,13 +18,13 @@ describe('Handle requests on other endpoints', () => {
   });
 
-  it('should return an error when a wrong url is provided', done => {
+  it('should return a welcome message for version 1 of the API', done => {
```
This test has been removed as well

daviddamilola committed:
feat(password-reset): add password reset endpoints

 - add test for password reset endpoint
 - add password reset endpoint routes.
 - add update Password service.
 - add verify email and send email Helper.
 - add password validation userSchema.
 - update documentation.

[Delivers #167749954]

hustlaviola commented on src/config/swagger.json:
```diff
@@ -98,6 +98,66 @@
+          }
+        ],
+        "responses": {
+          "201": {
```
You can change the status code here
```suggestion
          "200": {
```

hustlaviola commented on src/controllers/UserController.js:
```diff
@@ -55,4 +56,62 @@ export default class UserController {
+      id, email, createdAt, updatedAt
+    } = rowsAffected;
+    Responses.setSuccess(
+      201,
```
```suggestion
      200,
```

daviddamilola replied:
Thanks for the feedback @hustlaviola , changes have been implemented.

ldonjibson commented:
you still dey work on the stuff?
